quil-py/src/lib.rs

```diff
@@ -45,6 +45,66 @@ macro_rules! impl_to_quil {
     };
 }
 
+/// Implements pickling for an instruction by implementing __getstate__ and __reduce__.
```

Comment:
I know you link to the `__reduce__` documentation below, but it would probably be better to link to that here as well, or even instead of doing it on `__reduce__` itself.

Reply:
Updated the docstring per your comment.